src/pages/workspace/WorkspaceProfileAddressPage.tsx

```diff
@@ -0,0 +1,101 @@
+import React, {useCallback, useMemo, useState} from 'react';
+import AddressForm from '@components/AddressForm';
+import type {FormOnyxValues} from '@components/Form/types';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import ScreenWrapper from '@components/ScreenWrapper';
+import useLocalize from '@hooks/useLocalize';
+import Navigation from '@libs/Navigation/Navigation';
+import type {SettingsNavigatorParamList} from '@libs/Navigation/types';
+import {updateAddress} from '@userActions/Policy';
+import type {Country} from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type SCREENS from '@src/SCREENS';
+import type {CompanyAddress} from '@src/types/onyx/Policy';
+import type {WithPolicyProps} from './withPolicy';
+import withPolicy from './withPolicy';
+
+type WorkspaceProfileAddressPagePolicyProps = WithPolicyProps;
+
+type WorkspaceProfileAddressPageProps = StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.WORKSPACE.ADDRESS> & WorkspaceProfileAddressPagePolicyProps;
+
+function WorkspaceProfileAddressPage({policy}: WorkspaceProfileAddressPageProps) {
+    const {translate} = useLocalize();
+    const address = useMemo(() => policy?.address, [policy]);
+    const [currentCountry, setCurrentCountry] = useState(address?.country);
+    const [street1, street2] = (address?.addressStreet ?? '').split('\n');
+    const [state, setState] = useState(address?.state);
+    const [city, setCity] = useState(address?.city);
+    const [zipcode, setZipcode] = useState(address?.zip);
+
+    const updatePolicyAddress = (values: FormOnyxValues<typeof ONYXKEYS.FORMS.HOME_ADDRESS_FORM>) => {
+        if (!policy) {
+            return;
+        }
+        updateAddress(policy?.id, {
+            addressStreet: `${values.addressLine1?.trim() ?? ''}\n${values.addressLine2?.trim() ?? ''}`,
+            city: values.city.trim(),
+            state: values.state.trim(),
+            zip: values?.zipPostCode?.trim().toUpperCase() ?? '',
+            country: values.country,
+        });
+        Navigation.goBack();
+    };
+
+    const handleAddressChange = useCallback((value: unknown, key: unknown) => {
+        const countryValue = value as Country | '';
+        const addressKey = key as keyof CompanyAddress;
+
+        if (addressKey !== 'country' && addressKey !== 'state' && addressKey !== 'city' && addressKey !== 'zip') {
+            return;
+        }
+        if (addressKey === 'country') {
+            setCurrentCountry(countryValue);
+            setState('');
+            setCity('');
+            setZipcode('');
+            return;
+        }
+        if (addressKey === 'state') {
+            setState(countryValue);
+            setCity('');
+            setZipcode('');
+            return;
+        }
+        if (addressKey === 'city') {
+            setCity(countryValue);
+            setZipcode('');
+            return;
+        }
+        setZipcode(countryValue);
+    }, []);
```

Comment:
this will get simplified afterwards I guess

Reply:
Same as [here](https://github.com/Expensify/App/pull/38381#discussion_r1580776921).